### PR TITLE
Fixes for OCSP and CRL with non-blocking sockets

### PR DESCRIPTION
**Pull request by dgarske**
Fix for OCSP and CRL file descriptor check to allow 0.

### CODE REVIEW

File: src/crl.c [hunk 1]
```diff
@@ -349,7 +349,10 @@ int CheckCertCRL(WOLFSSL_CRL* crl, DecodedCert* cert)
         if (crl->crlIOCb) {
             ret = crl->crlIOCb(crl, (const char*)cert->extCrlInfo,
                                                         cert->extCrlInfoSz);
-            if (ret >= 0) {
+            if (ret == WOLFSSL_CBIO_ERR_WANT_READ) {
+                ret = WC_PENDING_E;
```
Comment (toddouska) [inline src/crl.c]: Isn't WC_PENDING specific to async crypto?  For nonblocking I/O issues we've been using WANT_READ.

File: src/ocsp.c [hunk 2]
```diff
@@ -445,6 +445,9 @@ int CheckOcspRequest(WOLFSSL_OCSP* ocsp, OcspRequest* ocspRequest,
         responseSz = ocsp->cm->ocspIOCb(ocsp->cm->ocspIOCtx, url, urlSz,
                                         request, requestSz, &response);
     }
+    if (responseSz == WOLFSSL_CBIO_ERR_WANT_READ) {
+        ret = WC_PENDING_E;
```
Comment (toddouska) [inline src/ocsp.c]: Why not WAN_READ instead of WC_PENDING_E?

File: src/crl.c (hunk 1 above)
Comment (dgarske) [inline src/crl.c]: Yeah this should be want read. I'll fix it up.

**Comment by kaleb-himes** [comment]
retest this please jenkins

**Pull request merged**